### PR TITLE
Add policy spec for task and project

### PR DESCRIPTION
**Pull request by Amir-Gaifullin**

### CODE REVIEW

File: spec/policies/task_policy_spec.rb [hunk 1]
```diff
@@ -35,9 +35,9 @@
     end
 
     context 'when user is not creator of the project' do
-      # let(:user) { User.new(id: 44) }
-      pending 'implement me'
-      # it { is_expected.to eq(false) }
+      let(:user) { User.new(id: 44) }
```
Comment (RasimKhusaenov) [inline spec/policies/task_policy_spec.rb:38]: Why you don't use Faker?

File: spec/policies/task_policy_spec.rb [hunk 2]
```diff
@@ -24,20 +24,21 @@
   describe '#create?' do
     subject { policy.create? }
 
-    let(:creator) { User.new(id: 43) }
-    let(:project) { Project.new(id: 42, user: user) }
-    let(:task) { Task.new(project: project) }
+    let(:project) { create :project, user_id: creator }
```
Comment (RasimKhusaenov) [inline spec/policies/task_policy_spec.rb:27]: You can use just `user`

File: spec/policies/task_policy_spec.rb [hunk 3]
```diff
@@ -24,20 +24,21 @@
...
-      let(:user) { creator }
+      let(:user) { create :user }
+      let(:creator) { user }
 
       it { is_expected.to be(true) }
     end
 
     context 'when user is not creator of the project' do
-      let(:user) { User.new(id: 44) }
+      let(:user) { create :user }
```
Comment (RasimKhusaenov) [inline spec/policies/task_policy_spec.rb:38]: You can move it outside context and DRY

File: spec/policies/task_policy_spec.rb [hunk 4]
```diff
@@ -24,20 +24,21 @@
   describe '#create?' do
     subject { policy.create? }
 
-    let(:creator) { User.new(id: 43) }
-    let(:project) { Project.new(id: 42, user: creator) }
-    let(:task) { Task.new(project: project) }
+    let(:project) { create :project, user_id: creator }
+    let(:task) { create :task, project: project, user_id: user }
```
Comment (RasimKhusaenov) [inline spec/policies/task_policy_spec.rb:28]: Why you need it at all? Your `create?` method is: https://github.com/Amir-Gaifullin/task-tracker/blob/9e034183583e0d8de677ec391d2c392d76258bbe/app/policies/task_policy.rb#L8-L10

**Review by RasimKhusaenov — CHANGES_REQUESTED** [review]
Maybe you should test all policy methods? And also your CI don't run your tests.

**Comment by Amir-Gaifullin** [comment]
@RasimKhusaenov re-request review (button dont work)

File: app/models/task.rb [hunk 5]
```diff
@@ -2,6 +2,7 @@ class Task < ApplicationRecord
   ALLOWED_STATUSES = %w[not_started started finished draft].freeze
 
   belongs_to :project
+  belongs_to :user, optional: true
```
Comment (RasimKhusaenov) [inline app/models/task.rb:5]: It is okay, but you don't use it. You can leave it out, but I prefer to code only what I need in PR.

File: spec/policies/project_policy_spec.rb [hunk 6]
```diff
@@ -33,23 +33,32 @@
...
   end
 
-  describe '#new?'
-  describe '#create?'
+  describe '#create?' do
+    subject { policy.create? }
 
-  describe '#edit?' do
-    pending 'implement me'
+    context 'when user is not authenticated' do
```
Comment (RasimKhusaenov) [inline spec/policies/project_policy_spec.rb:45]: I think, it is more correct to describe it like "present", not "authenticated".

File: spec/policies/task_policy_spec.rb [hunk 7]
```diff
@@ -24,20 +24,20 @@
...
-    let(:creator) { User.new(id: 43) }
-    let(:project) { Project.new(id: 42, user: creator) }
-    let(:task) { Task.new(project: project) }
+    let(:project) { create :project, user: creator }
+    let(:task) { create :task, project: project, user: user }
+    let(:user) { create :user }
 
     context 'when user is creator of the project' do
-      let(:user) { creator }
+      let(:creator) { user }
```
Comment (RasimKhusaenov) [inline spec/policies/task_policy_spec.rb:32]: Maybe it is more correct to code it like
```suggestion
      let(:user) { creator }
```
?

File: spec/policies/task_policy_spec.rb [hunk 8]
```diff
@@ -24,20 +24,20 @@
...
       it { is_expected.to be(true) }
     end
 
     context 'when user is not creator of the project' do
-      # let(:user) { User.new(id: 44) }
-      pending 'implement me'
-      # it { is_expected.to eq(false) }
+      let(:creator) { nil }
+
+      it { is_expected.to be(false) }
```
Comment (RasimKhusaenov) [inline spec/policies/task_policy_spec.rb:40]: What you think about this variant
```suggestion
      it { is_expected.to be_falsey }
```
?

**Comment by RasimKhusaenov** [comment]
@Amir-Gaifullin, it is because I am not collaborator in this repo. :)
You can invite me or leave it as is.

**Comment by RasimKhusaenov** [comment]
@Amir-Gaifullin it is for you :cupid:

File: app/models/task.rb (hunk 5 above)
Comment (Amir-Gaifullin) [reply app/models/task.rb:5]: u wrong 😢 🔫

Comment (RasimKhusaenov) [reply app/models/task.rb:5]: not u

Comment (Amir-Gaifullin) [reply app/models/task.rb:5]: ⚰️

**Comment by Amir-Gaifullin** [comment]
@RasimKhusaenov Approve ✅  or Ban ☠️

**Review by RasimKhusaenov — APPROVED** [review]

**Pull request merged**